### PR TITLE
Bug fix 1683359 - fluentd metrics not found

### PR DESCRIPTION
**Pull request by josefkarasek**

### CODE REVIEW

File: pkg/k8shandler/curation.go [hunk 1]
```diff
@@ -114,9 +114,9 @@ func (cluster *ClusterLogging) createOrUpdateCuratorConfigMap() error {
 		"curator",
 		cluster.Namespace,
 		map[string]string{
-			"actions.yaml":  string(utils.GetFileContents("files/curator-actions.yaml")),
```
Comment (ewolinetz) [inline pkg/k8shandler/curation.go]: if we do this then we can no longer use `make go-run`.

File: Dockerfile [hunk 2]
```diff
@@ -14,7 +14,7 @@ RUN INSTALL_PKGS=" \
     chmod og+w /tmp/_working_dir
 COPY --from=builder /go/src/github.com/openshift/cluster-logging-operator/_output/bin/cluster-logging-operator /usr/bin/
 COPY scripts/* /usr/bin/scripts/
-COPY files/ /usr/bin/files/
+COPY files/ /etc/clo/
```
Comment (ewolinetz) [inline Dockerfile]: what does this buy us? we will no longer have `files` under our `workdir`

**Comment by jcantrill** [comment]
@josefkarasek please split the bug fix referenced in this PR from the rules addition, which is a feature and currently not inclined to be accepted for 4.1 per @bparees

File: Dockerfile (hunk 2 above)
Comment (nhosoi) [reply Dockerfile]: I think this change is based on this review comment:
https://github.com/openshift/cluster-logging-operator/pull/112
Here's the new path:
fluentdAlertsFile = "/etc/clo/fluentd_prometheus_alerts.yaml"

**Comment by nhosoi** [comment]
This failure is real:
`ci/prow/e2e-operator — Job failed. `
In my manual testing, fluentd does not get installed.
```
$ oc get pods
NAME                                                  READY   STATUS    RESTARTS   AGE
cluster-logging-operator-744956f8c8-bkpxt             1/1     Running   0          10m
elasticsearch-clientdatamaster-0-1-66c7898d4b-vjzlj   2/2     Running   0          2m12s
elasticsearch-operator-54f5cc446f-6r4bm               1/1     Running   0          9m25s
kibana-7fb4fd4cc9-jm9x9                               2/2     Running   0          2m12s
```
Please note that there's no fluentd events reported.
```
$ oc get events | egrep fluent
$
```

**Comment by richm** [comment]
what about the cluster-logging-operator pod logs?

**Comment by nhosoi** [comment]
CLO reports lots of these errors...
```
time="2019-03-28T21:47:12Z" level=warning msg="Dropping key (openshift-logging/instance) out of the queue: Unable to create or update collection for \"instance\": Failure creating the fluentd ServiceMonitor: servicemonitors.monitoring.coreos.com is forbidden: User \"system:serviceaccount:openshift-logging:cluster-logging-operator\" cannot create resource \"servicemonitors\" in API group \"monitoring.coreos.com\" in the namespace \"openshift-logging\""
time="2019-03-28T21:47:30Z" level=error msg="error syncing key (openshift-logging/instance): Unable to create or update collection for \"instance\": Failure creating the fluentd ServiceMonitor: servicemonitors.monitoring.coreos.com is forbidden: User \"system:serviceaccount:openshift-logging:cluster-logging-operator\" cannot create resource \"servicemonitors\" in API group \"monitoring.coreos.com\" in the namespace \"openshift-logging\""
```

**Comment by richm** [comment]
@jcantrill @ewolinetz looks like clo needs additional rbac permissions to create the fluentd service monitor . . .

**Comment by nhosoi** [comment]
> @jcantrill @ewolinetz looks like clo needs additional rbac permissions to create the fluentd service monitor . . .
But it was not needed in @josefkarasek 's original pr/112...

**Comment by nhosoi** [comment]
Well, there was a similar problem 9 days ago in the previous PR and @josefkarasek fix it.  I should have paid more attention what was the cause... :(  @josefkarasek, could there be something similar in this PR, too?
```
9 days ago
time="2019-03-20T03:44:37Z" level=error msg="error syncing key (openshift-logging/example): Unable to create or update collection for \"example\": Failure creating the fluentd ServiceMonitor: servicemonitors.monitoring.coreos.com is forbidden: User \"system:serviceaccount:openshift-logging:cluster-logging-operator\" cannot create resource \"servicemonitors\" in API group \"monitoring.coreos.com\" in the namespace \"openshift-logging\""
```

File: pkg/k8shandler/curation.go (hunk 1 above)
Comment (josefkarasek) [reply pkg/k8shandler/curation.go]: I reverted this change

**Comment by ewolinetz** [comment]
Fluentd wasn't able to be created. Not a flake

**Comment by josefkarasek** [comment]
I had hoped this was gonna be enough - https://github.com/operator-framework/community-operators/commit/d7cf566847540e981e358b23d30ef5ecfd627b9f

Our tests install logging in a different way than users will in their environments?

> Fluentd wasn't able to be created. Not a flake

File: manifests/03-role.yaml [hunk 3]
```diff
@@ -78,6 +78,12 @@ rules:
   - "privileged"
   verbs:
   - use
+- apiGroups:
```
Comment (jcantrill) [inline manifests/03-role.yaml]: This has to be added to:
* https://github.com/openshift/cluster-logging-operator/blob/master/controller-manifests/cluster-logging.v4.1.0.clusterserviceversion.yaml
* https://github.com/operator-framework/community-operators/blob/master/community-operators/cluster-logging/cluster-logging.v4.1.0.clusterserviceversion.yaml

Comment (josefkarasek) [reply manifests/03-role.yaml]: ad 1 - in this PR
ad 2 - merged already https://github.com/operator-framework/community-operators/commit/d7cf566847540e981e358b23d30ef5ecfd627b9f

**Pull request merged**